test/correctness/simd_op_check_hvx.cpp

```diff
@@ -0,0 +1,664 @@
+        }
+    }
+    void add_tests() override {
+        Expr f32_1 = in_f32(x), f32_2 = in_f32(x+16), f32_3 = in_f32(x+32);
+        Expr f64_1 = in_f64(x), f64_2 = in_f64(x+16), f64_3 = in_f64(x+32);
+        Expr i8_1  = in_i8(x),  i8_2  = in_i8(x+16),  i8_3  = in_i8(x+32), i8_4 = in_i8(x + 48);
+        Expr u8_1  = in_u8(x),  u8_2  = in_u8(x+16),  u8_3  = in_u8(x+32), u8_4 = in_u8(x + 48);
+        Expr u8_even = in_u8(2*x), u8_odd = in_u8(2*x+1);
+        Expr i16_1 = in_i16(x), i16_2 = in_i16(x+16), i16_3 = in_i16(x+32);
+        Expr u16_1 = in_u16(x), u16_2 = in_u16(x+16), u16_3 = in_u16(x+32);
+        Expr i32_1 = in_i32(x), i32_2 = in_i32(x+16), i32_3 = in_i32(x+32);
+        Expr u32_1 = in_u32(x), u32_2 = in_u32(x+16), u32_3 = in_u32(x+32);
+        Expr i64_1 = in_i64(x), i64_2 = in_i64(x+16), i64_3 = in_i64(x+32);
+        Expr u64_1 = in_u64(x), u64_2 = in_u64(x+16), u64_3 = in_u64(x+32);
+        Expr bool_1 = (f32_1 > 0.3f), bool_2 = (f32_1 < -0.3f), bool_3 = (f32_1 != -0.34f);
+
+        int hvx_width = 0;
+        if (target.has_feature(Target::HVX_64)) {
+            hvx_width = 64;
+        } else if (target.has_feature(Target::HVX_128)) {
+            hvx_width = 128;
+        }
+
+        int isa_version;
+        if (target.has_feature(Halide::Target::HVX_v66)) {
+            isa_version = 66;
+        } else if (target.has_feature(Halide::Target::HVX_v65)) {
+            isa_version = 65;
+        } else if (target.has_feature(Halide::Target::HVX_v62)) {
+            isa_version = 62;
+        } else {
+            isa_version = 60;
+        }
+
+        // Verify that unaligned loads use the right instructions, and don't try to use
+        // immediates of more than 3 bits.
+        check("valign(v*,v*,#7)", hvx_width/1, in_u8(x + 7));
+        check("vlalign(v*,v*,#7)", hvx_width/1, in_u8(x + hvx_width - 7));
+        check("valign(v*,v*,r*)", hvx_width/1, in_u8(x + 8));
+        check("valign(v*,v*,r*)", hvx_width/1, in_u8(x + hvx_width - 8));
+        check("valign(v*,v*,#6)", hvx_width/1, in_u16(x + 3));
+        check("vlalign(v*,v*,#6)", hvx_width/1, in_u16(x + hvx_width - 3));
+        check("valign(v*,v*,r*)", hvx_width/1, in_u16(x + 4));
+        check("valign(v*,v*,r*)", hvx_width/1, in_u16(x + hvx_width - 4));
+
+        check("vunpack(v*.ub)", hvx_width/1, u16(u8_1));
+        check("vunpack(v*.ub)", hvx_width/1, i16(u8_1));
+        check("vunpack(v*.uh)", hvx_width/2, u32(u16_1));
+        check("vunpack(v*.uh)", hvx_width/2, i32(u16_1));
+        check("vunpack(v*.b)", hvx_width/1, u16(i8_1));
+        check("vunpack(v*.b)", hvx_width/1, i16(i8_1));
+        check("vunpack(v*.h)", hvx_width/2, u32(i16_1));
+        check("vunpack(v*.h)", hvx_width/2, i32(i16_1));
+
+        check("vunpack(v*.ub)", hvx_width/1, u32(u8_1));
+        check("vunpack(v*.ub)", hvx_width/1, i32(u8_1));
+        check("vunpack(v*.b)", hvx_width/1, u32(i8_1));
+        check("vunpack(v*.b)", hvx_width/1, i32(i8_1));
+
+#if 0
```

Comment:
I realize that the `#if 0` sections already existed, but now seems like a good time to open Issues for them (if they actually need to get dealt with eventually) or just deleted (if they don't)